packages/common/src/content/fetch.ts

```diff
@@ -24,21 +27,38 @@ export interface IFetchContentOptions extends IFetchItemAndEnrichmentsOptions {
   siteOrgKey?: string;
 }
 
+const maybeFetchLayerEnrichments = async (
+  itemAndEnrichments: IItemAndEnrichments,
+  options?: IFetchContentOptions
+) => {
+  // determine if this is a client-side feature layer view
+  const { item, data, layers } = itemAndEnrichments;
+  const layer =
+    layers && getItemLayer(item, layers, options && options.layerId);
+  // TODO: add recordCount here too?
```

Comment:
I'd prefer to fetch the record count here, where we fetch other enrichments, however, I need a lot of logic that currently only happens in `composeContent()` getters like `isProxied`, `url`, `viewDefinition`, so I'd have to turn all those getters into fns I could call here. So for now, I just fetch `recordCount` after calling `composeContent()`